galley/pkg/config/analysis/README.md

```diff
@@ -0,0 +1,189 @@
+# Analyzers
+
+The purpose of analyzers is to examine Istio configuration for potential problems that should be surfaced back to the user. An analyzer takes as input a Context object that contains methods to inspect the configuration snapshot being analyzed, as well as methods for reporting any issues discovered.
+
+## Writing Analyzers
+
+### 1. Create the code
+
+Analyzers need to implement the Analyzer interface ( in the `galley/pkg/config/analysis` package). They should be created under the analyzers subdirectory, and given their own appropriate subpackage.
+
+An annotated example:
+
+```go
+package virtualservice
+
+// <imports here>
+
+type GatewayAnalyzer struct{}
```

Comment:
Question: does a GatewayAnalyzer analyze all the possible invalid scenarios or only the scenario where the input VS has a Gateway that doesn't exist in the cluster? 
In that case, it is just a NIT, a more specific name like `NoGatewayAnalyzer` would help devs not put more than one responsibility per analyzer. I can imagine that there could be multiple scenarios to check for the gateways field in a VS.
For example, on Kiali we have all the different checkers split by responsibility: [Virtual Service checkers](https://github.com/kiali/kiali/tree/master/business/checkers/virtual_services). This way we apply Open/Close Principle.

Reply:
The analyzer in this doc is just intended as an example (although we do currently have an actual analyzer that's basically identical.) That said, if there are specific best practices we want to agree on around naming & organization, this doc will be a good place to record them.